src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -424,5 +425,52 @@ trait SubmissionComponent {
         where w.submission_id = ${submissionId}""".as[WorkflowInputResolutionListResult]
       }
     }
+
+    object DeleteSubmissionQuery extends RawSqlQuery {
+      val driver: JdbcDriver = SubmissionComponent.this.driver
+
+      def deleteAction(workspaceId: UUID) = {
+
+        def workspaceSubmissionQuery(workflow: SQLActionBuilder, join: SQLActionBuilder) = {
+          concatSqlActions(sql" inner join ", workflow, sql" w on ", join,
+            sql" inner join SUBMISSION s on w.submission_id=s.id where s.workspace_id=${workspaceId}")
+        }
+
+        val tables: Seq[SQLActionBuilder] = Seq(sql"WORKFLOW", sql"WORKFLOW_FAILURE")
+
+        DBIO.seq(
+          // delete submission attributes, workflow messages, and submission validations for workflows and
+          // workflow failures
+          concatSqlActions(
+            sql"""delete sa from SUBMISSION_ATTRIBUTE sa
+                  inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
+              """, workspaceSubmissionQuery(sql"WORKFLOW", sql"sv.workflow_id=w.id")).as[Int],
+          concatSqlActions(
+            sql"""delete sa from SUBMISSION_ATTRIBUTE sa
+                  inner join SUBMISSION_VALIDATION sv on sv.id=sa.owner_id
+              """, workspaceSubmissionQuery(sql"WORKFLOW_FAILURE", sql"sv.workflow_failure_id=w.id")).as[Int],
+          concatSqlActions(sql"delete wm from WORKFLOW_MESSAGE wm ",
```

Comment:
These are hard to read.  Could you rewrite these lines (452-461) as 5 calls to a method that takes the minimum possible input values: (name of table to delete, id column of table to delete, name to table to join on)?  The literal string interpolation syntax `#$` I mentioned above should help with this.

You may also be able to apply this to other calls inside `deleteAction` but I haven't thought that through yet.